src/components/AudioRecorderCompairUI/AudioRecorderCompairUI.js

```diff
@@ -31,6 +36,88 @@ export default class AudioRecorderCompair extends Component {
     });
   }
 
+  handleMic(){
+    if(!(this.pathname === 'practice')){
+      this.setState({ soundDetected: false, stopDetection: false });
+      this.startSoundDetection();
+      document.getElementById('startaudio_compair').click();
+    }
+    else{
+      document.getElementById('startaudio_compair').click();
+    }
+  }
+
+  handleStop() {
+    if(!(this.pathname === 'practice')){
+    document.getElementById('stopaudio_compair').click();
+    this.setState({ stopDetection: true });
+    }else{
+      document.getElementById('stopaudio_compair').click();
+    }
+  }
+
+  startSoundDetection = async () => {
```

Comment:
this should be separated from this audioRecoderCompaireUi.js

Reply:
not able to add this externally